Make trailing slashes optional

valpackett opened the pull request:
Now if you have a web server set up to handle this, you can set `trailing_slashes = false` in the config and use URLs without them.


## Code changes

* [x] Are you doing the PR on the `next` branch?
* [x] Have you created/updated the relevant documentation page(s)?

valpackett committed:
Make trailing slashes optional

Now if you have a web server set up to handle this, you can set
trailing_slashes = false in the config and use URLs without them.

sshine commented on docs/content/documentation/getting-started/configuration.md:
```diff
@@ -73,6 +73,11 @@ feed_filename = "atom.xml"
 
+# When set to "true", all links are generated with trailing slashes.
+# This is the default due to how most default web server configurations work.
+# DO NOT set this to false if you don't know what you're doing.
```
Unless "DO NOT ... if you don't know what you're doing." is a convention in Zola (I didn't check), perhaps it is better to inform what the consequence is. Since the consequence is both severe and relatable and can be explained briefly, how about:

```suggestion
# Trailing slashes change what relative paths (images, links) are relative to.
# Changing this may break all images and relative paths on a site, so be careful.
```

sshine reviewed (commented):
In reading the comments for this PR, I've learned that it makes perfect sense to make your relative paths default to the parent, even though this is not what's done historically or is the norm. It does make for nicer URLs where historically, the browser cooperated to remove trailing slashes to make the URLs nice.

I think the "due to how most default web server configurations work" argument warrants this.

And the argument for holding back should be to minimize the number of configurable parameters?

edwardmlyte committed:
docs: github action version updates (#1977)

* docs: github action version updates

* Uses latest version of the action
* Updated requirement for token to be set
  (https://github.com/shalzz/zola-deploy-action/blob/138a3568d1045f57bcdfec4b66ad37ac3dca5475/entrypoint.sh#L44
  will exit out without it)
* Reduced TOKEN scope and added scope to URL. Aligns with action docs

* docs: revert commentary on token requirement

* docs: add note about passing in automatic token

Keats committed:
Update themes gallery (#1999)

Co-authored-by: GitHub Action <action@github.com>

YesSeri committed:
Quotation mark bug fix (#2005)

Changed double quotation mark " to single quotation ', otherwise the code doesn't work.

berinaniesh committed:
update version of zola-deploy-action (#2012)

Keats committed:
Update themes gallery (#2014)

Co-authored-by: GitHub Action <action@github.com>
